Modules/Filtering/Convolution/include/itkFFTConvolutionImageFilter.h

```diff
@@ -181,18 +185,19 @@ class ITK_TEMPLATE_EXPORT FFTConvolutionImageFilter
   void
   CropOutput(InternalImageType * paddedOutput, ProgressAccumulator * progress, float progressWeight);
 
-  /** Get the lower bound for the padding of both the kernel and input
-   * images. Assuming that the regions of the kernel and input are the
-   * same, then this lower bound can be used to move the index of the
-   * padded kernel and padded input so that they are the same. This
-   * is important to avoid exceptions in filters that operate on these
-   * images. */
-  InputSizeType
-  GetPadLowerBound() const;
+  /** Get the radius of the kernel image. Used to pad the input image
+   *  for convolution. */
+  KernelSizeType
+  GetKernelRadius() const;
 
-  /** Get the pad size. */
-  InputSizeType
-  GetPadSize() const;
+  /** Get padding around the region of interest that results from FFT
+   *  factoring requirements. FFT typically requires that image side lengths
+   *  are factorable only by a fixed set of prime numbers (often 2, 3, and 5).
+   *  After the input image is padded for the kernel width and cropped to the
+   *  region of interest the result is then padded for FFT execution. This value
+   *  is reused for kernel padding and output cropping. */
+  InternalSizeType
+  GetFFTPadSize() const;
```

Comment:
Sorry to chime in. Thanks for the effort @tbirdso :100:. Now a petition: can the two getter methods above be exercised in the test, please?

Sames applies to any other method added (if any).

Thanks.

Reply:
Thank you for taking a look @jhlegarreta ! The two methods in question are used only internally, they are `protected` and not part of the public interface. I do not think that we typically test for these?

Reply:
I had quickly read the class and thought they were `public`. There are many such `protected` methods that are not covered as you mention. I think that is unfortunate. IMO, we should exercise them. The only way I can think of is creating a subclass in the tests for that purpose. If you happen to have a better solution, I'd like to hear about it. Thanks.

Reply:
Protected methods should be exercised indirectly, via the public methods which use them. I don't think we should specifically test them. The coverage can come naturally, via general testing of a class.

Reply:
While I agree with the sentiment of wanting to explicitly test protected methods, I'm uncertain that we have the resources to make this a common practice. Note that existing protected methods are shown to be covered via natural exercises in the existing [`itkFFTConvolutionImageFilter.h` coverage report](https://open.cdash.org/viewCoverageFile.php?buildid=7868666&fileid=49410952), which, while imperfect, has previously been noted as our standard for test coverage. Considering that we are not changing the `FFTConvolutionImageFilter` public interface I think it is okay to leave as-is for now.